geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/planning/guard/GraduatedQueryGuardTest.scala

```diff
@@ -0,0 +1,119 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.index.planning.guard
+
+import java.io.StringReader
+
+import com.typesafe.config.ConfigFactory
+import org.geotools.data.{Query, Transaction}
+import org.geotools.filter.text.ecql.ECQL
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.index.TestGeoMesaDataStore
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class GraduatedQueryGuardTest extends Specification {
+
+  "GraduatedQueryGuard" should {
+    "block queries with an excessive duration and spatial extent using graduated query guard" in {
+      // note: z3 needs to be declared first so it's picked for full table scans
+      val sft = SimpleFeatureTypes.createType("cea650aea6284b5281ee84c784cb56a7",
+        "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;geomesa.indices.enabled='z3,id,attr:name'")
+      // NB: Uses configuration in the test reference.conf
+      sft.getUserData.put("geomesa.query.interceptors",
+        "org.locationtech.geomesa.index.planning.guard.GraduatedQueryGuard")
+
+      val ds = new TestGeoMesaDataStore(true)
+      ds.createSchema(sft)
+
+      val valid = Seq(
+        "name = 'bob'",
+        "IN('123')",
+        "bbox(geom,0,0,.2,.4) AND dtg during 2020-01-01T00:00:00.000Z/2020-02-01T00:00:00.000Z",
+        // Three Corner cases.
+        // Note that these periods are under the limit by two seconds.
+        "bbox(geom,0,0,1,1) AND dtg during 2020-01-01T00:00:00.000Z/P60D",
+        "bbox(geom,0,0,2,5) AND dtg during 2020-01-01T00:00:00.000Z/P3D",
+        "bbox(geom,-180,-90,180,90) AND dtg during 2020-01-01T00:00:00.000Z/P1D",
+        "bbox(geom,0,0,2,4) AND dtg during 2020-01-01T00:00:00.000Z/2020-01-02T00:00:00.000Z",
+        "bbox(geom,-10,-10,10,10) AND dtg during 2020-01-01T00:00:00.000Z/2020-01-01T23:00:00.000Z",
+        "bbox(geom,-10,-10,10,10) AND (dtg during 2020-01-01T00:00:00.000Z/2020-01-01T00:59:59.000Z OR dtg during 2020-01-01T12:00:00.000Z/2020-01-01T12:59:59.000Z)"
+      )
+
+      val invalid = Seq(
+        "INCLUDE",
+        "bbox(geom,-10,-10,10,10)",
+        "bbox(geom,-180,-90,180,90)",
```

Comment:
If you update this PR at all, could you add a time range filter to this list?

Reply:
Maybe a dtg after, and a dtg during?

Reply:
sure

Reply:
which case did you mean, exactly? you already had during and after examples in the list.

Reply:
oh you mean without a bbox?

Reply:
Yes!